firedrake/slate/slac/kernel_builder.py

```diff
@@ -128,6 +130,84 @@ def dimension(e):
         self.coefficient_vecs = coeff_vecs
         self._setup()
 
+    @cached_property
+    def terminal_flops(self):
+        flops = 0
+        nfacets = self.expression.ufl_domain().ufl_cell().num_facets()
+        for ctx in self.context_kernels:
+            itype = ctx.original_integral_type
+            for k in ctx.tsfc_kernels:
+                kinfo = k.kinfo
+                if itype == "cell":
+                    flops += kinfo.kernel.num_flops
+                elif itype.startswith("interior_facet"):
+                    # Executed once per facet (approximation)
+                    flops += kinfo.kernel.num_flops * nfacets
```

Comment:
Just noting here that this is a bit tricky if the integral is defined over facet sub-domains. In which case, not all facets will be included in the local computation.

Reply:
Yes, there's no real way to get this exactly right, short of runtime counting in some way.